dagobah/core/job.py

```diff
@@ -312,13 +312,13 @@ def _complete_task(self, task_name, **kwargs):
             self.backend.release_lock()
 
         if kwargs.get('success', None) is False:
-            task = self.tasks[task_name]
+            task = self.tasks_snapshot[task_name]
             try:
                 self.backend.acquire_lock()
                 if self.event_handler:
-                    self.event_handler.emit('task_failed',
-                                            task._serialize(
-                                                include_run_logs=True))
+                    task_data = task._serialize(include_run_logs=True)
+                    task_data['delimiter'] = self.JIJ_DELIM
```

Comment:
Since the data is not held in a config, and the email functions hve no knowledge of any of the core classes, I decided it would be best if the delimiter was added on just before emission of the event.

Another alternative might be to make it an optional arg in self._serialize(), thoughts?

I chose not to because I thought it was an unnecessary signature change, but I'm not married to this decision